.github/workflows/pre-commit.yaml

```
@@ -0,0 +1,32 @@
+name: Pre-commit linters and code checks
+
+on:
+  pull_request:
+    paths-ignore:
+      - 'README.md'
+  push:
+    branches:
+      - develop
+    paths-ignore:
+      - 'README.md'
+
+jobs:
+  pre_commit:
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Set up Python
+      uses: actions/setup-python@v3
+      with:
+        python-version: 3.8
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pre-commit==3.3.3
+
+    # Currently pre-commit is run for all staged files.
+    # In the future we can run using --all-files
```

May be we could make it a TODO:?

```suggestion
    # TODO: In the future we can run using --all-files
```

---

Good suggestion. I added the TODO items.